jetty-webapp/src/main/java/org/eclipse/jetty/webapp/WebInfConfiguration.java

```diff
@@ -456,7 +460,10 @@ public void resolveTempDirectory(WebAppContext context)
             File work = new File(jettyBase, "work");
             if (work.exists() && work.isDirectory() && work.canWrite())
             {
-                context.setPersistTempDirectory(true);
+                if (context.getServer().isWorkDirectoryPersistent())
+                {
+                    context.setPersistTempDirectory(true);
+                }
```

Comment:
Maybe we should also do this with an attribute so we don't commit to a forever API:

```suggestion
                Object workNotPersistent = context.getServer().getAttribute(WORK_DIRECTORY_NOT_PERSISTENT);
                if (workNotPersistent != null && Boolean.valueOf(workNotPersistent))
                 context.setPersistTempDirectory(true);
```